Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -266,20 +290,23 @@ protected void OnStalling(ChainedBlock chainedBlock)
 
             if (behavior != null)
             {
-                behavior.QualityScore = Math.Max(MinQualityScore, behavior.QualityScore - 1);
-                if (behavior.QualityScore == MinQualityScore)
+                double penalty = this.peerQuality.CalculateNextBlockTimeoutQualityPenalty();
```

Comment:
yes I think this is good think in general - the original design had this and I think it was good decision exactly for the reason you've mentioned - the very next block is somehow special and it is easy to imagine an edge case when we could be completely stuck without it